src/dbnode/client/write_state.go

```diff
@@ -199,6 +237,16 @@ func (w *writeState) completionFn(result interface{}, err error) {
 	w.decRef()
 }
 
+func (w *writeState) setHostSuccessList(hostID, pairedHostID string) {
```

Comment:
nit: add `...WithLock` suffix to method names when you're relying on holding a lock to mutate state